Bottom: ea9e14be5b079ab8803fbc90426e30b20baa488c
Top:    f2d1bcb7abcce3c91fb5eda092717c8e6676a00a
Author: Seth Vidal <skvidal@fedoraproject.org>
Date:   2011-08-10 14:40:50 -0400

if we're using --update we make repo cache dirs in /var/tmp

clean them up before we go


---

diff --git a/createrepo/__init__.py b/createrepo/__init__.py
index 61f6d66..f1894f5 100644
--- a/createrepo/__init__.py
+++ b/createrepo/__init__.py
@@ -1195,9 +1195,17 @@ class MetaDataGenerator:
                     raise MDError, msg
 
         self._cleanup_tmp_repodata_dir()
+        self._cleanup_update_tmp_dir()        
         self._write_out_read_pkgs_list()
 
 
+    def _cleanup_update_tmp_dir(self):
+        if not self.conf.update:
+            return
+        
+        shutil.rmtree(self.oldData._repo.basecachedir, ignore_errors=True)
+        shutil.rmtree(self.oldData._repo.base_persistdir, ignore_errors=True)
+        
     def _write_out_read_pkgs_list(self):
         # write out the read_pkgs_list file with self.read_pkgs
         if self.conf.read_pkgs_list:
